src/lib/Microsoft.Health.Fhir.Ingest.Legacy/Errors/ExceptionContextExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.EventHubs;
+
+namespace Microsoft.Health.Events.Errors
+{
+    public static class ExceptionContextExtensions
+    {
+        private const string EventsAttr = "RelatedEvents";
+
+        public static void AddEventContext(this Exception ex, IEnumerable<EventData> events)
+        {
+            ex.Data[EventsAttr] = events;
```

Comment:
Why do we need to add in the list of events to the `Exception`? Where does that get used? Also, are we sure that this attribute won't get logged with the Exception??

Reply:
The original code was added as part of the Event Messaging service work @wi-y is doing so we can capture the context of the events that caused an error.  This is deployed and happening today and isn't automatically logged in the exception.

It is currently used in the class MeasurementEventNormalizationService and the method ProcessErrorAsync.  Previously this code was pointing to [src/lib/Microsoft.Health.Events/Errors/IErrorMessageWithEvents.cs](https://github.com/microsoft/iomt-fhir/pull/231/files/288c8cbe7544d15d79a916de4cae85f7650f6562#diff-7f7122b7defeb4d195cd3d421c1cc8e318c16a22cb4b583f8f62d221e2e2bd8f).  This is no longer valid since this class was updated to remove the old event hub SDK in favor of the new one.  The updated extension in the Microsoft.Health.Events library still consumes EventData but the type is different since it comes from the new library, not the old one.

The other option is I can remove this extension and the corresponding code in the class MeasurementEventNormalizationService
@wi-y do you have any preferences?  The class MeasurementEventNormalizationService is no longer used outside the legacy functions.

Reply:
We don't really need it since we don't really plan on supporting error message service for the function based app. We can remove this and the corresponding code in MeasurementEventNormalizationService

Reply:
Thanks I will update to remove this piece.